src/pybind/mgr/stats/fs/perf_stats.py

```diff
@@ -201,7 +202,7 @@ def notify_fsmap(self):
         #Reregister the user queries when there is a new rank0 mds
         with self.lock:
             gid_state = FSPerfStats.get_rank0_mds_gid_state(self.module.get('fs_map'))
-            if not gid_state:
+            if not gid_state or not self.mx_last_updated:
```

Comment:
Does this really help? Could you explain why?

AFAICT, initializing `self.mx_last_updated = 0.0` in `FSPerfStats.__init__` should suffice. @joscollin @neesingh-rh

Reply:
There's no need to take it further, no need to start the timer too. So catch it and return early.

Reply:
> There's no need to take it further, no need to start the timer too. So catch it and return early.

That's explaining what the change is and not why the change is required. What can go wrong if the timer kicks in ? In `re_register_queries`:

```
if self.mx_last_updated >= ua_last_updated:
    ....
```

The else path would be taken which would register the timer again. Are you worried about registering the timer over and over again when `self.mx_last_updated == 0`?

Reply:
So, the additional check wasn't really required? What about my query in the above comment @joscollin @neesingh-rh

Reply:
As you mentioned [here](https://github.com/ceph/ceph/pull/56525#discussion_r1546065807), the initialisation is sufficient for resolving this issue. The additional check is not required. The timer usually starts whenever there's an fsmap change, that's normal.